Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridComboBoxColumn.cs

```diff
@@ -0,0 +1,457 @@
+
+                    dataItem.GetType().GetProperty(Binding.Path.Path).SetValue(dataItem, newValue);
+                }
+            };
+
+            return comboBox;
+        }
+
+        /// <summary>
+        /// Gets a read-only <see cref="T:System.Windows.Controls.TextBlock"/> element that is bound to the column's <see cref="P:Microsoft.Toolkit.Uwp.UI.Controls.DataGridBoundColumn.Binding"/> property value.
+        /// </summary>
+        /// <param name="cell">The cell that will contain the generated element.</param>
+        /// <param name="dataItem">The data item represented by the row that contains the intended cell.</param>
+        /// <returns>A new, read-only <see cref="T:System.Windows.Controls.TextBlock"/> element that is bound to the column's <see cref="P:Microsoft.Toolkit.Uwp.UI.Controls.DataGridBoundColumn.Binding"/> property value.</returns>
+        protected override FrameworkElement GenerateElement(DataGridCell cell, object dataItem)
+        {
+            var textBlockElement = new TextBlock
+            {
+                Margin = new Thickness(DATAGRIDCOMBOBOXCOLUMN_leftMargin, 0.0, DATAGRIDCOMBOBOXCOLUMN_rightMargin, 0.0),
+                VerticalAlignment = VerticalAlignment.Center
+            };
+
+            if (DependencyProperty.UnsetValue != ReadLocalValue(DataGridComboBoxColumn.FontFamilyProperty))
+            {
+                textBlockElement.FontFamily = this.FontFamily;
+            }
+
+            if (_fontSize.HasValue)
+            {
+                textBlockElement.FontSize = _fontSize.Value;
+            }
+
+            if (_fontStyle.HasValue)
+            {
+                textBlockElement.FontStyle = _fontStyle.Value;
+            }
+
+            if (_fontWeight.HasValue)
+            {
+                textBlockElement.FontWeight = _fontWeight.Value;
+            }
+
+            RefreshForeground(textBlockElement, (cell != null & cell.OwningRow != null) ? cell.OwningRow.ComputedForeground : null);
+
+            bool isDesignMode = Windows.ApplicationModel.DesignMode.DesignModeEnabled;
+
+            if (this.Binding != null || !isDesignMode)
+            {
+                textBlockElement.SetBinding(TextBlock.TextProperty, this.Binding);
+            }
+
+            return textBlockElement;
+        }
+
+        /// <summary>
+        /// Causes the column cell being edited to revert to the specified value.
+        /// </summary>
+        /// <param name="editingElement">The element that the column displays for a cell in editing mode.</param>
+        /// <param name="uneditedValue">The previous, unedited value in the cell being edited.</param>
+        protected override void CancelCellEdit(FrameworkElement editingElement, object uneditedValue)
+        {
+            var comboBox = editingElement as ComboBox;
+            if (comboBox != null)
+            {
+                var value = uneditedValue.GetType().GetProperty(Binding.Path.Path).GetValue(uneditedValue);
```

Comment:
uneditedValue may be null, right?

Reply:
You are right, uneditedValue may be null and has to be checked for.